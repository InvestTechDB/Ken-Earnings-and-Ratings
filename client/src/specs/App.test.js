import React from 'react';
import { shallow, mount, render } from 'enzyme';
import App from './app.js';

describe('component: App', () => {
  it('should render initial layout', () => {
    let wrapper = shallow(<App />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('should not have any preexisting data loaded for the app', () => {
    let wrapper = shallow(<App />);
    let loadState = wrapper.state().exampleDataLoaded;
    expect(loadState).toEqual(false);
  });

  it('should load data from the database once component was loaded', () => {
    let wrapper = shallow(<App />);
    setTimeout(() => {
      let loadState = wrapper.state().exampleDataLoaded;
      expect(loadState).toEqual(true);
    }, 0);
  });
});
