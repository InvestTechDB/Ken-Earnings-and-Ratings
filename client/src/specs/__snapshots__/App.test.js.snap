// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: App should render initial layout 1`] = `
Array [
  <div>
    <Summary
      buySummary=""
      ratings={Array []}
      sellSummary=""
    />
    <br />
    <Earnings
      actualEarnings={Array []}
      className="earningsBlock"
      estimatedEarnings={Array []}
    />
  </div>,
]
`;
